### PR TITLE
Analytics Consent

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Adds analytics consent popup in SDK App

Screenshot 2024-07-10 at 11 36 46 AM

## How is this patch tested? If it is not, please explain why.

Locally

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
  - Added Google Analytics (GA) tracking functionality.
  - Introduced `AnalyticsConsent` component for managing user consent regarding analytics tracking.

- **Enhancements**
  - Updated the structure of the `DatasetPage` for improved clarity and layout.
  - Expanded the GitHub Actions workflow to trigger on the `py-panels-develop` branch. 

- **Bug Fixes**
  - Streamlined the `Nav` component by removing unnecessary analytics integration.

- **Chores**
  - Restricted the version of `setuptools` for better compatibility.

### CODE REVIEW

File: app/packages/app/src/components/AnalyticsConsent.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      doNotTrack: doNotTrack,
+      debug: info.dev,
+    });
+  }, [doNotTrack, info, ready, setAnalyticsInfo]);
+
+  if (!show) {
+    return null;
+  }
+
+  return (
```
Comment (brimoor) [inline app/packages/app/src/components/AnalyticsConsent.tsx:68]: @benjaminpkane can you share a screenshot in light mode to ensure it looks good there too?

File: app/packages/app/src/components/AnalyticsConsent.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        alignItems="center"
+        borderTop={(theme) => `1px solid ${theme.palette.divider}`}
+        backgroundColor="background.paper"
+      >
+        <Grid padding={2}>
+          <Typography variant="h6" marginBottom={1}>
+            Help us improve FiftyOne
+          </Typography>
+          <Typography marginBottom={1}>
+            We use our own and third party cookies to understand how you use
```
Comment (brimoor) [inline app/packages/app/src/components/AnalyticsConsent.tsx:82]: suggested copy tweak:

```
            We use cookies to understand how FiftyOne is used and to improve
            the product. You can help us by enabling analytics.
```

**Review by brimoor — CHANGES_REQUESTED** [review]

**Comment by benjaminpkane** [comment]
Updated copy + light mode
Screenshot 2024-07-11 at 11 19 56 AM

**Review by brimoor — APPROVED** [review]
LGTM from a product perspective. Defer to eng for implementation

File: app/packages/app/src/components/Analytics.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        dev
+        doNotTrack
+        uid
+        version
+      }
+    `,
+    fragment
+  );
+
+  useGA(false, info);
```
Comment (imanjra) [inline app/packages/app/src/components/Analytics.tsx:47]: Is unconditional false intentional here?

File: app/packages/app/src/components/AnalyticsConsent.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        </Grid>
+      </Grid>
+    </PinBottom>
+  );
+}
+
+// a component that pins the content to the bottom of the screen, floating
+function PinBottom({ children }: React.PropsWithChildren) {
+  return (
+    <Box position="fixed" bottom={0} width="100%" zIndex={51}>
```
Comment (imanjra) [inline app/packages/app/src/components/AnalyticsConsent.tsx:106]: 5️⃣ 1️⃣

File: app/packages/app/src/pages/datasets/DatasetPage.tsx [hunk 5]
```diff
@@ -122,8 +121,7 @@ const DatasetPage: Route<DatasetPageQuery> = ({ prepared }) => {
           </datasetQueryContext.Provider>
         )}
       </div>
-      <Snackbar />
```
Comment (imanjra) [inline app/packages/app/src/pages/datasets/DatasetPage.tsx:125]: Is the removal of `Snackbar` intentional?

**Review by imanjra — APPROVED** [review]
Left some questions but LGTM

File: app/packages/app/src/pages/datasets/DatasetPage.tsx (hunk 5 above)
Comment (benjaminpkane) [reply app/packages/app/src/pages/datasets/DatasetPage.tsx:125]: Unintentional 🤦 Thanks!

File: app/packages/app/src/components/Analytics.tsx (hunk 3 above)
Comment (benjaminpkane) [reply app/packages/app/src/components/Analytics.tsx:47]: Unintentional 🙇 will fix

**Pull request merged**